src/cider/nrepl/middleware/debug.clj

```diff
@@ -577,6 +592,25 @@ this map (identified by a key), and will `dissoc` it afterwards."}
                         (reset! *skip-breaks* old-breaks#)))))
         `(expand-break ~form ~dbg-state ~original-form)))))
 
+(defmacro breakpoint-if-exception
+  "Wrap form in a try-catch that has breakpoint on exception.
+  Ignores uninteresting forms.
+  Uninteresting forms are symbols that resolve to `clojure.core`
+  (taking locals into account), and sexps whose head is present in
+  `irrelevant-return-value-forms`. Used as :breakfunction in `tag-form`."
+  [form dbg-state original-form]
+  (if (uninteresting-form? &env form)
+    form
+    `(try ~form
+          (catch Exception ex#
+            (let [exn-message# (.getMessage ex#)
+                  break-result# (expand-break exn-message# ~dbg-state ~original-form)]
```

Comment:
It would also be nice if we could overload the `s`->`:stacktrace` and `p`->`:inspect` commands here, 
using Cider's stacktrace analyser to make better sense of the stacktrace, or the inspector to traverse the ex-info map.

Currently the result being returned here is only `exn-message#`, so calling the :inspect command simply inspects the error string. 

Maybe a better method is to return the exception object `ex#` itself? On the client end we could process the response to pretty-print the message alone as an overlay. Inspecting would then work automatically, and we could add an condition in `debug-stacktrace` to analyse the current value if `(instance? Throwable value)`